AnnotationModule.py

```diff
@@ -0,0 +1,63 @@
+import csv
+import xml.etree.ElementTree as ET
+import cv2
+import numpy as np
+import openslide
+import os
+from skimage import io
+from BaseImage import printMaskHelper
+from PIL import Image, ImageDraw
+
+def getPoints(xmlfile):
+    # create element tree object
+    tree = ET.parse(xmlfile)
+
+    # get root element
+    root = tree.getroot()
+    annCount = 0
+
+    # create empty list for annotations
+    annotations = []
+
+    for annotation in root.findall('Annotation'):
+        for regions in annotation.findall('Regions'):
+            for region in regions.findall('Region'):
+                for vertices in region.findall('Vertices'):
+                    points[annCount] = [None] * len(vertices.findall('Vertex'))
+                    k = 0
+                    for vertex in vertices.findall('Vertex'):
+                        points[annCount][k] = (int(float(vertex.get('X'))), int(float(vertex.get('Y'))))
+                        k = k + 1
+                annCount = annCount + 1
+
+    return points
+
+def maskOutAnnotation(s,params):
+
+    annotationPath = params.get("annotation_path",s["dir"])
+    annotationPath = annotationPath + os.sep + s["filename"].rsplit(".",1)[0] + ".xml"
+
+    points = getPoints(annotationPath)
+
+    downsize_factor = int(s["height"])/s["img_mask_use"].shape[0]
+
+    points = [(a[0][0] / downsize_factor, a[0][1] / downsize_factor) for a in zip(points)]
+
+    img = Image.new('L', (s["img_mask_use"].shape[1], s["img_mask_use"].shape[0]), 0)

```

Comment:
is PIL already a HistoQC requirement, or does this approach require adding on another dependency?

Reply:
PIL is a new dependency.